automation_oca/models/automation_configuration_activity.py

```diff
@@ -0,0 +1,226 @@
+    )
+    domain = fields.Char(
+        required=True, default="[]", help="Filter to apply specifically"
+    )
+    applied_domain = fields.Char(
+        compute="_compute_applied_domain",
+        recursive=True,
+    )
+    parent_id = fields.Many2one("automation.configuration.activity", ondelete="cascade")
+    model_id = fields.Many2one(related="configuration_id.model_id")
+    model = fields.Char(related="model_id.model")
+    child_ids = fields.One2many(
+        "automation.configuration.activity", inverse_name="parent_id"
+    )
+    activity_type = fields.Selection(
+        [("mail", "Mail"), ("action", "Server Action")], required=True, default="mail"
+    )
+    trigger_interval_hours = fields.Integer(
+        compute="_compute_trigger_interval_hours", store=True
+    )
+    trigger_interval = fields.Integer()
+    trigger_interval_type = fields.Selection(
+        [("hours", "Hours"), ("days", "Days")], required=True, default="hours"
+    )
+    expiry = fields.Boolean()
+    expiry_interval = fields.Integer()
+    expiry_interval_type = fields.Selection(
+        [("hours", "Hours"), ("days", "Days")], required=True, default="hours"
+    )
+    trigger_type = fields.Selection(
+        [
+            ("start", "start of workflow"),
+            ("activity", "execution of another activity"),
+            ("mail_open", "Mail opened"),
+            ("mail_not_open", "Mail not opened"),
+            ("mail_reply", "Mail replied"),
+            ("mail_not_reply", "Mail not replied"),
+            ("mail_click", "Mail clicked"),
+            ("mail_not_clicked", "Mail not clicked"),
+            ("mail_bounce", "Mail bounced"),
+        ],
+        required=True,
+        default="start",
+    )
+    mail_author_id = fields.Many2one(
+        "res.partner", required=True, default=lambda r: r.env.user.id
+    )
+    mail_template_id = fields.Many2one(
+        "mail.template", domain="[('model_id', '=', model_id)]"
+    )
+    server_action_id = fields.Many2one(
+        "ir.actions.server", domain="[('model_id', '=', model_id)]"
+    )
+    parent_position = fields.Integer(
+        compute="_compute_parent_position", recursive=True, store=True
+    )
+    graph_data = fields.Json(compute="_compute_graph_data")
+    graph_done = fields.Integer(compute="_compute_total_graph_data")
+    graph_error = fields.Integer(compute="_compute_total_graph_data")
+
+    @api.onchange("trigger_type")
+    def _onchange_trigger_type(self):
+        if self.trigger_type == "start":
+            self.parent_id = False
+
+    ########################################
+    # Graph computed fields ################
+    ########################################
+
+    @api.depends()
+    def _compute_graph_data(self):
+        total = self.env["automation.record.activity"].read_group(
+            [
+                ("configuration_activity_id", "in", self.ids),
+                ("processed_on", ">=", fields.Date.today() + relativedelta(days=-14)),
```

Comment:
avoid issues with countries with UTC-n (LATAM)
```suggestion
                ("processed_on", ">=", fields.Date.context_today(self) + relativedelta(days=-14)),
```

Reply:
Fixed